webwolf/src/main/java/org/owasp/webwolf/mailbox/Email.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of WebGoat, an Open Web Application Security Project utility. For details, please see http://www.owasp.org/
+ *
+ * Copyright (c) 2002 - 2019 Bruce Mayhew
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *
+ * Getting Source ==============
+ *
+ * Source for this application is maintained at https://github.com/WebGoat/WebGoat, a repository for free software projects.
+ */
+
+package org.owasp.webwolf.mailbox;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * @author nbaars
+ * @since 8/20/17.
+ */
+@Data
+@Builder
+@AllArgsConstructor
+@Entity
+@NoArgsConstructor
+public class Email implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @JsonIgnore
+    private LocalDateTime time = LocalDateTime.now();
+    @Column(length = 1024)
+    private String contents;
+    private String sender;
+    private String title;
+    private String recipient;
+
+    public String getSummary() {
+        return "-" + this.contents.substring(0, Math.min(50, contents.length()));
+    }
+
+    public LocalDateTime getTimestamp() {
+        return time;
+    }
+
+    public String getTime() {
+        return DateTimeFormatter.ofPattern("h:mm a").format(time);
+    }
+
+    public String getShortSender() {
+        return sender.substring(0, sender.indexOf("@"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 71 does not find the substring and returns -1, the call to substring() on line 71 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string sender.